### PR TITLE
feat: query suggestions

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

File: src/actions/query.js [hunk 1]
```diff
@@ -356,14 +365,41 @@ function msearch(
...
 
-function appbaseSearch(
+function getQuerySuggestionsId(componentId) {
+	return `${componentId}__suggestions`;
+}
+
+function appbaseSearch({
 	query,
 	orderOfQueries,
 	appendToHits = false,
```
Comment (bietkul) [inline src/actions/query.js:375]: @ShahAnuj2610 Do you think that default values will work in that way?
If yes then share an example.
It seems wrong to me.

File: src/actions/query.js [hunk 2]
```diff
@@ -689,14 +748,17 @@ export function executeQuery(
...
-					finalQuery,
+				dispatch(appbaseSearch({
+					query: finalQuery,
 					orderOfQueries,
-					false,
-					componentId.endsWith('__internal'),
-					undefined,
+					appendToHits: false,
+					isInternalComponent: componentId.endsWith('__internal'),
+					appendToAggs: undefined,
```
Comment (bietkul) [inline src/actions/query.js]: @ShahAnuj2610 It's redundant now.

File: src/actions/query.js [hunk 3]
```diff
@@ -689,14 +748,17 @@ export function executeQuery(
...
 			if (isAppbaseEnabled) {
-				dispatch(appbaseSearch(
-					finalQuery,
+				dispatch(appbaseSearch({
+					query: finalQuery,
 					orderOfQueries,
-					false,
-					componentId.endsWith('__internal'),
-					undefined,
+					appendToHits: false,
```
Comment (bietkul) [inline src/actions/query.js]: @ShahAnuj2610 redundant

File: src/actions/query.js [hunk 4]
```diff
@@ -828,7 +890,13 @@ export function loadMore(component, newOptions, appendToHits = true, appendToAgg
 				...getDependentQueries(getState(), component, []),
 			};
 			const finalQuery = Object.keys(appbaseQuery).map(c => appbaseQuery[c]);
-			dispatch(appbaseSearch(finalQuery, [component], appendToHits, false, appendToAggs));
+			dispatch(appbaseSearch({
+				query: finalQuery,
+				orderOfQueries: [component],
+				appendToHits,
+				isInternalComponent: false,
```
Comment (bietkul) [inline src/actions/query.js]: @ShahAnuj2610 It seems redundant.

**Review by bietkul — CHANGES_REQUESTED** [review]
@ShahAnuj2610 You made a breaking change in `appbaseSearch` method. Did you update all references for it?

File: src/actions/query.js (hunk 1 above)
Comment (ShahAnuj2610) [reply src/actions/query.js:375]: @bietkul didn't get you. Can you share example for which it would break?

**Comment by ShahAnuj2610** [comment]
> @ShahAnuj2610 You made a breaking change in `appbaseSearch` method. Did you update all references for it?

@bietkul yes, I've updated all the references

**Review by bietkul — APPROVED** [review]

**Review by bietkul — APPROVED** [review]

**Pull request merged**